packages/contracts-bedrock/test/setup/DeployVariations.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Testing utilities
+import { CommonTest } from "test/setup/CommonTest.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract DeployVariations_Test is CommonTest {
+    function setUp() public override {
+        // prevent calling the base CommonTest.setUp() function
+    }
+
+    // Enable features which should be possible to enable or disable regardless of other options.
+    function enableAddOns(bool _enableCGT, bool _enableAltDa) public {
+        if (_enableCGT) {
+            ERC20 token = new ERC20("Silly", "SIL");
+            super.enableCustomGasToken(address(token));
+        }
+        if (_enableAltDa) {
+            super.enableAltDA();
+        }
+    }
+
+    function testFuzz_enableFaultProofs(bool _enableCGT, bool _enableAltDa) public virtual {
+        enableAddOns(_enableCGT, _enableAltDa);
+        super.enableFaultProofs();
```

Comment:
Are the `super.` calls intended to just not revert? Also seems the test don't match up with the flags?

Reply:
> Are the super. calls intended to just not revert? 

Correct. There are many places within the deployment that can fail, this is basically a smoke test for those.

> Also seems the test don't match up with the flags?

How so? I added some natspec on the tests, lmk if something else is missing.

Reply:
Ah ok, I was thinking we'd need a `testFuzz_enableCGT` for example, but I see what you're doing here now.